src/main/java/sg/ncl/MainController.java

```diff
@@ -602,83 +625,24 @@ public String editAccountDetails(@ModelAttribute("editUser") User2 editUser, fin
                 credHeaders.setContentType(MediaType.APPLICATION_JSON);
                 credHeaders.set("Authorization", AUTHORIZATION_HEADER);
 
-                HttpEntity<String> credRequest = new HttpEntity<String>(credObject.toString(), headers);
-                restTemplate.exchange(properties.getSioCredUrl() + session.getAttribute("id"), HttpMethod.PUT, credRequest, String.class);
-                redirectAttributes.addFlashAttribute("editPassword", "success");
+                HttpEntity<String> credRequest = new HttpEntity<>(credObject.toString(), headers);
+                restTemplate.setErrorHandler(new MyResponseErrorHandler());
+                ResponseEntity response = restTemplate.exchange(properties.getSioCredUrl() + session.getAttribute("id"), HttpMethod.PUT, credRequest, String.class);
+                String responseBody = response.getBody().toString();
+
+                try {
+                    if (RestUtil.isError(response.getStatusCode())) {
+                        MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+                        redirectAttributes.addFlashAttribute("editPassword", "fail");
+                    } else {
+                        redirectAttributes.addFlashAttribute("editPassword", "success");
+                    }
+                } catch (IOException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)